staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -264,7 +264,7 @@ func (g *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error)
 	providerID := node.Spec.ProviderID
 	if providerID == "" {
 		var err error
-		if providerID, err = g.InstanceID(ctx, types.NodeName(node.Name)); err != nil {
```

Comment:
I think InstanceID itself is expected to return the InstanceNotFound error so there's a potential optimization here to return `false, nil` early if InstanceID returned InstanceNotFound.

Reply:
done